File: 014/sol.cpp
#include <iostream>

int main() {
    const int n = 1000000;
    long temp, count, res;
    long max = 0;
    
    for (int i = 2; i < n; i++) {
        count = 0;
        temp = i;
        while (temp > 1) {
            if (temp % 2 == 0) {
                temp /= 2;
            } else {
                temp = (3 * temp) + 1;
            }
            count++;
        }
        if (count > max) {
            max = count;
            res = i;
        }
    }
    std::cout << res;
}

File: 025/sol.py
from itertools import count
from functools import cache

@cache
def fib(n):
    if n < 2:
        return n
    return fib(n-1) + fib(n-2)

def solve():
    for n in count(1):
        if len(str(fib(n))) >= 1000:
            res = n
            break
    return res

if __name__ == "__main__":
    print(solve())
